pkg/apis/migration/v1alpha1/condition.go

```diff
@@ -83,12 +91,42 @@ func (r *Conditions) DeleteCondition(cndTypes ...string) {
 	if r.Conditions == nil {
 		return
 	}
+	kept := []Condition{}
```

Comment:
What's the purpose of the `DeleteCondition` method now? I thought we were clearing our the entire set when reconcile starts?

Reply:
The staged/commit work-flow is optional and mainly used by the _validation.go_ code.  There are still cases where the explicit set/delete work-flow is used.  For example: the condition related to ensuring the BSL/DSL in the plan controller.  And, in some cases, the final `Ready` condition.

Reply:
ACK